src/Rule/Email.php

```diff
@@ -93,6 +94,11 @@ protected function validateValue($value, DataSetInterface $dataSet = null): Resu
             }
         }
 
+        if ($this->enableIDN && $valid === false) {
+            $pattern = '/^([a-zA-Z0-9._%+-]+)@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.)|(([\w-]+\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\]?)$/';
```

Comment:
That could be moved to a customizable property same as other regex patterns.

Reply:
@samdark there is a private pattern regex, maybe a new property $patternIdnEmail as a public property, is ok?

Reply:
@samdark i upated the PR ;)

Reply:
@samdark is it ok?

Reply:
Yes.